core/types/block.go

```diff
@@ -286,7 +287,7 @@ func NewBlock(header *Header, body *Body, receipts []*Receipt, hasher TrieHasher
 		b.header.Bloom = CreateBloom(receipts)
 	}
 
-	if len(uncles) == 0 {
+	if len(uncles) == 0 && !bType.IsMigratedChain() {
```

Comment:
Do we maybe want the EmptyUncleHash for new blocks in a migrated chain that aren't the genesis?

Reply:
Probably we do 👍

Reply:
Well spotted, looks like that could have broken future block generation, but strange that it didn't cause any tests to fail.